src/pip/_internal/commands/install.py

```diff
@@ -226,6 +228,30 @@ def add_options(self):
     @with_cleanup
     def run(self, options, args):
         # type: (Values, List[str]) -> int
+        # Warn if running pip with sudo
+        try:
+            if sys.platform == 'darwin' or sys.platform == 'linux':
+                is_sudo = os.getuid() == 0
+            elif sys.platform == 'win32' or sys.platform == 'cygwin':
+                is_sudo = ctypes.windll.shell32.IsUserAnAdmin() != 0
+        except AttributeError:
+            is_sudo = False
+
+        is_venv = (hasattr(sys, 'real_prefix')
+                   or (hasattr(sys, 'base_prefix')
+                       and sys.base_prefix != sys.prefix))
```

Comment:
https://github.com/pypa/pip/blob/7369ac2ee85815d4a059c667cc415f2000df7ed5/src/pip/_internal/utils/virtualenv.py#L37-L41

Reply:
Changed to `if is_sudo and not running_under_virtualenv():`